src/evaluators/JSXElement.js

```diff
@@ -288,13 +286,18 @@ function evaluateJSXAttributes(
     } catch (e) {
       if (realm.isInPureScope() && e instanceof FatalError) {
         let flagProps = hasNoPartialKeyOrRef(realm, config);
-
+        let emptyObject = Create.ObjectCreate(realm, realm.intrinsics.ObjectPrototype);
+        // TODO: maybe we can use temporalAlias and/or snapshots to improve this?
         config = AbstractValue.createTemporalFromBuildFunction(
           realm,
           ObjectValue,
-          [objAssign, config, ...abstractPropsArgs],
+          [objAssign, emptyObject, ...abstractPropsArgs],
```

Comment:
This seems like change that fixes something. Is there a test case for this?

Reply:
No, this doesn't fix anything. Purely changed whilst I was looking at all Object.assign sites.

Reply:
Crying out for a test case. ;-)

Reply:
I don't really know what either of you expect me to test here? This is purely a cosmetic change.

Reply:
If this fixes nothing and you are sure that the current version works, you should put this in its own PR with appropriate comments. Ideally, you would like to see a test case that a) mutates config and then fails and b) recovers and then does the right thing at runtime.

Reply:
I'll pull it out of this PR, but the reason why it's safe is because we scrap the old config and use the **return** value from `Object.assign`, not the `target` argument.

Reply:
Done

Reply:
Will do.